airflow-core/src/airflow/models/variable.py

```diff
@@ -302,7 +302,7 @@ def update(
             )
 
     @staticmethod
-    def delete(key: str, session: Session | None = None) -> int:
+    def delete(key: str, session: Session | None = None) -> None | int:
```

Comment:
This because in older version (v2), it returned the count of vars (1 or 0) 
Newer one returns nothing,

Reply:
For sake of easier types, why not have this always return `1` on the compat path?

Reply:
Will do